internal/servicename.go

```diff
@@ -21,27 +21,66 @@
 package internal
 
 import (
-	"fmt"
+	"errors"
 	"regexp"
 )
 
-var (
-	// Must starts with [a-z].
-	// Can contain [a-z] and 0-9] with non-consecutive dashes.
-	validNameRegex = regexp.MustCompile("^[a-z]+([a-z0-9]|[^-]-)*[^-]$")
+// We disallow UUIDs explicitly, though they're otherwise valid patterns.
+var uuidRegex = regexp.MustCompile("[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}")
 
-	// We disallow UUIDs explicitly (they would be accepted by validNameRegex alone)
-	uuidRegex = regexp.MustCompile("[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}")
-)
-
-// ValidateServiceName returns an error if the given name is not a valid
-// service name.
+// ValidateServiceName returns an error if the given servive name is invalid.
+// Valid names are at least two characters long, start with [a-z], contain only
+// [0-9a-z] and non-consecutive hyphens, and end in [0-9a-z].
 func ValidateServiceName(name string) error {
-	if !validNameRegex.MatchString(name) {
-		return fmt.Errorf("service name must begin with a letter and consist only of dash-delimited lower-case ASCII alphanumeric words")
+	if len(name) < 2 {
```

Comment:
You could be cute and combine all of these with `multierr.Combine` instead of returning error after each one, this would also give all validation errors back to the caller, just be careful that a later validation doesn't rely on an earlier one (ie there would be a panic in a later validation), this wouldn't be good anyways though.